src/ca.c

```diff
@@ -419,6 +301,12 @@ unsigned char dh_g[256] =
 
 int dvbca_close_device(ca_device_t *c);
 static int ca_send_datetime(ca_device_t *d);
+int populate_resources(ca_device_t *d, int *resource_ids);
+int ca_write_apdu_session(ca_device_t *d, int session_number, int resource,
+                          const void *data, int len);
+int ca_write_apdu(ca_device_t *d, int resource, const void *data, int len);
+int find_session_for_resource(int resource);
+int asn_1_decode(int *length, unsigned char *asn_1_array);
```

Comment:
IMO to keep this file more managable in size, ASN.1 functions could be moved to a different file

Reply:
Same thing with the AES and other crypto stuff in this file, but maybe it should all be moved in a later PR.

Reply:
Once I get it working I can split to multiple files and fix the magic numbers